### PR TITLE
Prevent bokeh from eating scroll events if wheel tool isn't active

### LINKED ISSUE #1450: wheel zoom interaction improvements
It's a little unintuitive when you have a big plot, no zoom happens and window doesn't scroll either.

---

This now marked `feature`:

* [x] do not capture mouse events when tool is inactive
* [ ] capability to have tool, but not have it enabled by default

### PR DESCRIPTION
**Pull request by birdsarah**
Partially fixes #1450.

This ensures that scroll events are only absorbed if wheelzoomtool is active. I've tested this manually and tested some other plots too and seems to do what we want.

@mistakevin @mattpap @bryevdv, I'd love some pointers on unit testing this. 

I thought this would be a good candidate for an integration / selenium test, so I set up a few. I just started with some basic interaction and I was able to hover, select glyphs and pan, which I'm pretty happy about. (You can see the code on the branch: https://github.com/bokeh/bokeh/blob/feature/1450_test_wheel_zoom_interactions/).  However, I then discovered that selenium has nothing to trigger a `mousewheel` event. I started trying to trigger via javascript

```js
     var e = Bokeh.$.Event( "wheel", { deltaY: 100, deltaX: 100 } );
     Bokeh.$('.bk-canvas-events').trigger(e)
```
https://github.com/bokeh/bokeh/blob/feature/1450_test_wheel_zoom_interactions/tests/integration/file/test_wheel_zoom_interactions.py#L62

I haven't got deep into it yet, but thought I'd ask for any thoughts @mistakevin, @mattpap, @bryevdv 

Maybe this PR can have some unit tests, maybe it can go in as is.

### CODE REVIEW

**Comment by mistakevin** [comment]
@birdsarah 
Here is a unit test I came up with using cheerio and mocha-jsdom https://github.com/bokeh/bokeh/pull/2718
Is this a direction we like?

**Comment by birdsarah** [comment]
@mistakevin thanks so much for this. cheerio and jsdom seem awesome - tests are super easy to read and understand.

The changes that seem to have been necessary to ui_events.coffee seem to be for testing rather than making the code more testable (if that makes sense as a distinction) which doesn't seem ideal. But I  definitely don't have any better suggestions as I'm new to these testing libraries. And having tests is better than not having tests as we can then feel more confident to refactor again if we subsequently find a more elegant solution.

**Comment by birdsarah** [comment]
@mistakevin I updated the tests to check the _mouse_wheel behavior and then made some small refactors. Can you take a look and tell me what you think?

**Comment by birdsarah** [comment]
@mistakevin - I think this is finally ready for review - many, many thanks for your help with the testing. 

Hoping this can make it into 0.9.3.

ping @mattpap @bryevdv @damianavila @canavandl

File: bokehjs/test/test_common/test_ui_events.coffee [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    deltaX: 100
+    e.bokeh = {}
+
+    @preventDefault = sinon.spy(e, "preventDefault")
+    @stopPropagation = sinon.spy(e, "stopPropagation")
+    @e = e
+
+    @ui_event = new ui_events({
+      hit_area: $('canvas')
+      tool_manager: gestures: scroll: active: true
```
Comment (mattpap) [inline bokehjs/test/test_common/test_ui_events.coffee]: Is this a deeply nested object? If so, would you mind rewriting this more explicitly.

**Comment by birdsarah** [comment]
Thanks @mattpap - I think it's more readable now. (I'm new to coffeescript so not sure what's obvious and isn't to people who are more comfortable with it.)

**Comment by mattpap** [comment]
> I'm new to coffeescript so not sure what's obvious (...)

We were supposed to create a style policy to avoid confusion, but this didn't happen yet.

As you mention you're new to this language, one more suggestion. Avoid using `f 1` syntax instead of `f(1)` as much as possible (there is only one good case for parenthesis-less syntax, where an argument is a multi-line function).

**Comment by mistakevin** [comment]
+1 for a style policy

**Comment by canavandl** [comment]
As a new-to-coffeescript contributor, i'm also +1 on the style policy. Even if it's "Use this other person's style policy".

This seems popular: https://github.com/polarmobile/coffeescript-style-guide

**Comment by birdsarah** [comment]
@mattpap, many thanks for this, it's very helpful. In fact I was bitten by this when writing this PR as I had assumed I could do `@_hammer_element` to call a function instead of `@_hammer_element()`.

Can you please check my latest commit: 9b6d90fa49a78ffa0c01889bd2ee29a7f388bbc1

Also to clarify, looking around it seems that the convention is:

` expect(build_views.jQueryUIPrefixer el).to.equal "bar-ui-in-the-middle"`
and 
`utils = require "../utils"`

I'm assuming these are also common exceptions to the `()` rule so I haven't updated those lines with parens.

**Comment by mistakevin** [comment]
Wrote an issue related to a style guide https://github.com/bokeh/bokeh/issues/2766

**Comment by damianavila** [comment]
OK, this seems good to me...

**Comment by fpliger** [comment]
LGTM too .. but tests are failing locally. I'm afraid is due to some spurious errors due to my own env rather then a real issue. If it's looking good for others I'm +1 about merging.

**Comment by damianavila** [comment]
>but tests are failing locally. I'm afraid is due to some spurious errors due to my own env rather then a real issue.

If you remove node_modules and `npm install` before testing, it should be OK... so merging this now...

**Comment by mattpap** [comment]
> (...) `utils = require "../utils"`. I'm assuming these are also common exceptions (...)

Not at all. Parenthesis should be used here as well. In `equals` case it would make sense to drop parenthesis, if you could drop the dot as well, like in Scala, where single argument method can act like an operator. Sadly and not unexpectedly this is not possible in CoffeeScript, so parenthesis should be used as with any other function call.

**Comment by birdsarah** [comment]
@mattpap, this is interesting, thanks. And I will follow your suggestions as I keep working on bokeh. I'm glad that there's an issue open to get a coffeescript style guide (#2766)

**Pull request merged**